tests/test_channel_builder.py

```diff
@@ -26,15 +29,20 @@ def test_compose2():
     updatedMsg = updateMsg(rules, msg)
 
     channels = updatedMsg['body']['channels']
-    assert len(channels) == 4
+    assert len(channels) == 5
 
     assert 'sms_republished_v1_neo' in channels
     assert 'ody_engine_fps_id' in channels
     assert 'foo' in channels
     assert 'bar' in channels
+    assert 'subscriber_republished_3' in channels
 
 
 def test_some_empty_fields():
+    import random
+
+    random.seed(0)
```

Comment:
I don't know how this test harness works, so I'm going to assume it's not multithreaded and won't mess up your random seed here.